app/services/site_configs/update_navigation.rb

```diff
@@ -0,0 +1,53 @@
+module SiteConfigs
+  class UpdateNavigation
+    class NavigationLink
+      include ActiveModel::Validations
+
+      URI_REGEXP = URI::DEFAULT_PARSER.make_regexp(%w[http https]).freeze
```

Comment:
This regex could be better, e.g. `https://` is valid according to it since almost all parts are optional.